spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -28,27 +28,57 @@ class DebuggingWidget(RichJupyterWidget):
     Spyder
     """
 
+    def __init__(self, *args, **kwargs):
+        super(DebuggingWidget, self).__init__(*args, **kwargs)
+        self._previous_prompt = None
+        self._input_queue = []
+        self._input_ready = False
+
+    def monkeypatch_kernel_client(self, client):
```

Comment:
I don't like this name too much (monkey patching is not how things are done in Python, but sub-classing).

What this is essentially doing is replacing the kernel client's `input`, so what if we avoid the inner function definition (`monkeypatched_input`), call it `set_queued_input` and use it like this in `set_kernel_client_and_manager`

```python
self.kernel_client = kernel_client
if self.kernel_client is not None:
    self.set_queued_input(self.kernel_client)
```

?